Mathlib/AlgebraicGeometry/Morphisms/Constructors.lean

```diff
@@ -242,6 +251,134 @@ lemma MorphismProperty.topologically_propertyIsLocalAtTarget
 
 end Topologically
 
+/-- `stalkwise P` holds for a morphism if all stalks satisfy `P`. -/
+def stalkwise
+    (P : ∀ {R S : Type u} [CommRing R] [CommRing S], (R →+* S) → Prop) :
+    MorphismProperty Scheme.{u} := fun _ _ f => ∀ x, P (PresheafedSpace.stalkMap f.val x)
+
+section Stalkwise
+
+variable {P : ∀ {R S : Type u} [CommRing R] [CommRing S], (R →+* S) → Prop}
+
+/-- If `P` respects isos, then `stalkwise P` respects isos. -/
+lemma stalkwise_respectsIso (hP : RingHom.RespectsIso P) :
+    (MorphismProperty.stalkwise P).RespectsIso where
+  left {X Y Z} e f hf := by
+    simp only [stalkwise, Scheme.comp_coeBase, TopCat.coe_comp, Function.comp_apply]
+    intro x
+    erw [PresheafedSpace.stalkMap.comp]
+    exact (RingHom.RespectsIso.cancel_right_isIso hP _ _).mpr <| hf (e.hom.val.base x)
+  right {X Y Z} e f hf := by
+    simp only [stalkwise, Scheme.comp_coeBase, TopCat.coe_comp, Function.comp_apply]
+    intro x
+    erw [PresheafedSpace.stalkMap.comp]
+    exact (RingHom.RespectsIso.cancel_left_isIso hP _ _).mpr <| hf x
+
+/-- If `P` respects isos, then `stalkwise P` is local at the target. -/
+lemma stalkwiseIsLocalAtTarget_of_respectsIso (hP : RingHom.RespectsIso P) :
+    PropertyIsLocalAtTarget (MorphismProperty.stalkwise P) := by
+  have hP' : (RingHom.toMorphismProperty P).RespectsIso :=
+    RingHom.toMorphismProperty_respectsIso_iff.mp hP
+  apply propertyIsLocalAtTarget_of_morphismRestrict
+  · exact stalkwise_respectsIso hP
+  · intro X Y f U hf x
+    apply (hP'.arrow_mk_iso_iff <| morphismRestrictStalkMap f U x).mpr <| hf _
+  · intro X Y f ι U hU hf x
+    have hy : f.val.base x ∈ iSup U := by rw [hU]; trivial
+    obtain ⟨i, hi⟩ := Opens.mem_iSup.mp hy
+    exact (hP'.arrow_mk_iso_iff <| morphismRestrictStalkMap f (U i) ⟨x, hi⟩).mp <| hf i ⟨x, hi⟩
+
+end Stalkwise
+
+section Restriction
+
+/-- If `P` is a property of scheme morphisms, we may restrict `P` to morphisms with affine target
+to obtain an `AffineTargetMorphismProperty`. -/
+def toAffineTargetMorphismProperty (P : MorphismProperty Scheme) :
```

Comment:
Maybe `AffineTargetMorphismProperty.of` so that you can use `targetAffineLocally (.of P)`.

Reply:
Like so? `targetAffineLocally (.of P)` this notation does not seem to work though.